src/main/java/TaskList.java

```diff
@@ -0,0 +1,178 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class to handle all the tasks.
+ */
+public class TaskList {
+    private static final String ADD_TODO_COMMAND = "todo";
+    private static final String ADD_DEADLINE_COMMAND = "deadline";
+    private static final String ADD_EVENT_COMMAND = "event";
+
+    List<Task> lst = new ArrayList<>();
+
+    /**
+     * Iterates and returns enumerated list of tasks.
+     */
+    public void iterateList() {
+        System.out.println("Here are the items in your list:");
+        int i = 1;
+        for (Task item : lst) {
+            System.out.println(i + ". " + item.toString());
+            i++;
+        }
+    };
+
+    public List<Task> fetchTasks() {
+        return lst;
+    }
+
+    /**
+     * Checks for task in the given index and marks it as completed if valid.
+     *
+     * @param str task number given one-based indexing
+     */
+    public void markAsDone(String str) {
+        try {
+            int index = Integer.parseInt(str) - 1;
+            Task task = lst.get(index);
+            task.markAsDone();
+
+            Ui.formatText();
+            System.out.println("Sweet! I have marked the following task as done:\n"
+                    + task.toString());
+            Ui.formatText();
+        } catch (final NumberFormatException e) {
+            System.err.println("Oof, did you type a valid number or not?");
+        } catch (IndexOutOfBoundsException exception) {
+            System.err.println("You don't have so many items, dumbass!");
+        }
+    }
+
+    /**
+     * Adds task to list of tasks if it is a valid input.
+     * @param input user input in command line
+     */
+    public void addTask(String input) {
```

Comment:
Would it be better to split the logic into different methods for better organisation?

Reply:
I agree with @jlxw48, perhaps you could further abstract this method?